drv/lpc55-update-server/src/images.rs

```diff
@@ -26,79 +44,515 @@ extern "C" {
 // Location of the NXP header
 pub const HEADER_BLOCK: usize = 0;
 
-// NXP LPC55's mixed header/vector table offsets
-const RESET_VECTOR_OFFSET: usize = 0x04;
-pub const LENGTH_OFFSET: usize = 0x20;
-pub const HEADER_OFFSET: u32 = 0x130;
-const MAGIC_OFFSET: usize = HEADER_OFFSET as usize;
+// An image may have an ImageHeader located after the
```

Comment:
**question:** Is this presence decided by eg. version of the header, where older images don't have the header? Might be useful information here.

Reply:
The early versions of the bootloader did not have the header and we have had discussions in the past about removing the need for the header. `update_server` tolerates a missing ImageHeader.